.github/workflows/codestyle.yml

```diff
@@ -29,3 +33,74 @@ jobs:
 
       - name: Run Black
         run: black --check tardis
+  ruff: 
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.12.3' 
```

Comment:
Do we keep this in track with TARDIS?

Reply:
Yes,
https://github.com/tardis-sn/tardis/blob/0df98ce0c9bf392c3e1b2406c27b4faa6ad07060/conda-lock.yml#L6522